sdk/storage/azure-storage-files-shares/src/share_directory_client.cpp

```diff
@@ -197,6 +197,118 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
       throw;
     }
   }
+  Azure::Response<ShareFileClient> ShareDirectoryClient::RenameFile(
+      const std::string& fileName,
+      const std::string& destinationFilePath,
+      const RenameFileOptions& options,
+      const Azure::Core::Context& context) const
+  {
+    auto sourceFileUrl = m_shareDirectoryUrl;
+    sourceFileUrl.AppendPath(_internal::UrlEncodePath(fileName));
+
+    auto destinationFileUrl = m_shareDirectoryUrl;
+    destinationFileUrl.AppendPath(_internal::UrlEncodePath(destinationFilePath));
+
+    auto protocolLayerOptions = _detail::FileClient::RenameFileOptions();
+    protocolLayerOptions.RenameSource =  sourceFileUrl.GetAbsoluteUrl();
+    protocolLayerOptions.ReplaceIfExists = options.ReplaceIfExists;
+    protocolLayerOptions.IgnoreReadOnly = options.IgnoreReadOnly;
+    protocolLayerOptions.DestinationLeaseId = options.AccessConditions.LeaseId;
+    protocolLayerOptions.SourceLeaseId = options.SourceAccessConditions.LeaseId;
+    protocolLayerOptions.FileContentType = options.FileContentType;
+    protocolLayerOptions.Metadata
+        = std::map<std::string, std::string>(options.Metadata.begin(), options.Metadata.end());
+    if (!options.SmbProperties.Attributes.GetValues().empty())
+    {
+      protocolLayerOptions.FileAttributes = options.SmbProperties.Attributes.ToString();
+    }
```

Comment:
Do we have to check empty here? If we don't have to, convenience layer code would be more concise.